PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -191,21 +191,20 @@ function Get-PCStorageDiagnosticInfo
     #
     # Makes a list of cluster nodes - filtered for if they are available, and hides the different options
 	# passing cluster/nodes to the script
-    #
+    #	
 
 	$FilteredNodelist = @()
 
 	Function GetFilteredNodeList {
 		if ($FilteredNodelist.Count -eq 0)
 		{
+			$NodesToPing = @();
+			
 			if ($Nodelist -ne "")
```

Comment:
$Nodelist.Count (its a declared array in the params, so its safe/proper). Its not comparable to an empty string; at least, depending on whatever its doing here is tricky. Powershell typechecking/willingness to do blind type conversion is painfully obtuse sometimes, this is one of them.